packages/app/src/runner/unifiedRunner.ts

```diff
@@ -35,17 +47,34 @@ export function useUnifiedRunner () {
           return
         }
 
-        if (queryFile === RUN_ALL_SPECS_KEY) {
-          return specStore.setActiveSpec(RUN_ALL_SPECS)
+        const activeSpecInSpecsList = queryFile === RUN_ALL_SPECS_KEY
+          ? RUN_ALL_SPECS
+          : specs.value.find((x) => x.relative === queryFile)
+
+        if (isEqual(route.query, prevQuery) && isEqual(activeSpecInSpecsList, specStore.activeSpec)) {
+          return
+        }
+
+        prevQuery = route.query
+
+        if (!activeSpecInSpecsList) {
+          return specStore.setActiveSpec(null)
         }
 
-        const activeSpecInSpecsList = specs.value.find((x) => x.relative === queryFile)
+        if (route.query.runId) {
+          const res = await testsForRunMutation.executeMutation({ runId: route.query.runId as string })
 
-        if (activeSpecInSpecsList && specStore.activeSpec?.relative !== activeSpecInSpecsList.relative) {
-          specStore.setActiveSpec(activeSpecInSpecsList)
-        } else if (!activeSpecInSpecsList) {
-          specStore.setActiveSpec(null)
+          // spread to get rid of readonly
+          const failedTests = [...res.data?.testsForRun || []]
+
+          specStore.setTestFilter(failedTests.length ? failedTests : undefined)
+        } else {
+          specStore.setTestFilter(undefined)
         }
+
+        // Either there is a new spec or the runId has changed. Returning a new object
```

Comment:
when would the runId change? when the cloud data updates? `runId` is a cloud concept correct?

Reply:
The data management of the current `runId` is still in-progress, but the end goal is that a user could be looking at a given run when a new run is kicked off by CI or another user. A user can move to view the newer run using an in-UI transition which will update the `runId` and kick off new queries

Reply:
can we document this information?